src/libraries/System.Text.Json/tests/Serialization/Value.ReadTests.SpecializedCollections.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Specialized;
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public static partial class ValueTests
+    {
+        [Fact]
+        public static void Read_SpecializedCollection()
+        {
+            BitVector32 bv32 = JsonSerializer.Deserialize<BitVector32>(@"{""Data"":4}");
+            // Data property is skipped because it doesn't have a setter.
+            Assert.Equal(0, bv32.Data);
+
+            HybridDictionary hd = JsonSerializer.Deserialize<HybridDictionary>(@"{""key"":""value""}");
+            Assert.Equal(1, hd.Count);
+            Assert.Equal("value", ((JsonElement)hd["key"]).GetString());
+
+            IOrderedDictionary iod = JsonSerializer.Deserialize<OrderedDictionary>(@"{""key"":""value""}");
+            Assert.Equal(1, iod.Count);
+            Assert.Equal("value", ((JsonElement)iod["key"]).GetString());
+
+            ListDictionary ld = JsonSerializer.Deserialize<ListDictionary>(@"{""key"":""value""}");
+            Assert.Equal(1, ld.Count);
+            Assert.Equal("value", ((JsonElement)ld["key"]).GetString());
+        }
+
+        [Fact]
+        public static void Read_SpecializedCollection_Throws()
+        {
+            //// Add method for this collection only accepts strings, even though it only implements IList which usually
+            //// indicates that the element type is typeof(object).
+            //Assert.Throws<InvalidCastException>(() => JsonSerializer.Deserialize<StringCollection>(@"[""1"", ""2""]"));
+
+            //// Not supported. Not IList, and we don't detect the add method for this collection.
+            //Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<StringDictionary>(@"[{""Key"": ""key"",""Value"":""value""}]"));
+
+            //// Int key is not allowed.
+            //Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<HybridDictionary>(@"{1:""value""}"));
```

Comment:
nit: commented out tests. This commit has a clean up for this and the other change I mentioned above: https://github.com/layomia/dotnet_runtime/commit/39797f3ed94d7913577950598d72c2c1a0c08e4b.

Reply:
OK thanks I'll apply it